### ISSUE TITLE
Implement dynamic dates for azure IP range URL

### ISSUE THREAD

**Issue by pnadolny13** [meltano/squared#354]
Discussed in slack https://meltano.slack.com/archives/C01UTUSP34M/p1660163864060499.

The azure url is hard coded and needs to be manually updated via a PR every 14 days. Find a way to dynamically generate the date and inject it into the job so we don't need to do this anymore.

**Comment by pnadolny13** [comment]
@aaronsteers @edgarrmondragon any ideas on this? I cant figure out a way to do this.

My preferred approach is to:
- write a python script that outputs a date to a file
- cat that file into an env var
- use that env var in my [config](https://github.com/meltano/squared/blob/817fb132058750a8c2a99e175dbb176e3078a83f/data/extract/extractors.meltano.yml#L43) instead of the hardcoded date like `pattern: ServiceTags_Public_${SSA_AZURE_FILE_DATE}.json`
- call my python script prior to the tap like `meltano run my_python_script_util tap-spreadsheets-anywhere target-snowflake`

It doesnt look like the env var gets resolved because I see `ServiceTags_Public_${SSA_AZURE_FILE_DATE}.json` inside my `TAP_SPREADSHEETS_ANYWHERE_TABLES` environment variable using debug mode. Is this because env vars arent resolved in settings that low? Its a value in a list of dictionary settings within my config.

**Comment by aaronsteers** [comment]
> It doesnt look like the env var gets resolved because I see `ServiceTags_Public_${SSA_AZURE_FILE_DATE}.json` inside my `TAP_SPREADSHEETS_ANYWHERE_TABLES` environment variable using debug mode.

Can you confirm this is still the case with the latest version of Meltano from yesterday?

**Comment by aaronsteers** [comment]
@pnadolny13 - I'd start with a simple test that you're able to pass env vars into the config value:

````
export SSA_AZURE_FILE_DATE=2022-01-01
meltano ...
```

With strict mode on, you should get an error if `$SSA_AZURE_FILE_DATE` is not defined.

**Comment by aaronsteers** [comment]
To the meta point, it could be interesting to have a `meltano-env-seed-ext` extension that could seed Meltano config based on zero or more input values.

```bash
meltano invoke meltano-env-seed  # executes `custom_env_seed.py` and sends results to `.env` file or another setting store
meltano run...                   # runs with the customized settings
```

**Comment by pnadolny13** [comment]
>I don't know how much value this adds beyond regular linux shell scripting though. 🤔

@aaronsteers I would like that! The value for me is that I dont need to figure out how to implement that date logic in different shells, CI and prod look to be slightly different and python is probably preferred for most meltano users. Also I run in prod using jobs so usually I'd be able to do `SSA_AZURE_FILE_DATE=20220808 meltano run tap-spreadsheets-anywhere target-snowflake` but in prod I cant do that using the Airflow dag generator/jobs. Does that make sense?

**Comment by pnadolny13** [comment]
>@pnadolny13 - I'd start with a simple test that you're able to pass env vars into the config value:

@aaronsteers yeah that doesnt work for me still, I saw this in a recent slack thread https://meltano.slack.com/archives/C01TCRBBJD7/p1661448119512469?thread_ts=1661421990.054269&cid=C01TCRBBJD7, it sounds like I can't use env vars in a nested setting like that. right? Seems like the option would be to pass in the whole `tables` array as an env var which is pretty clunky 😞.

**Comment by pnadolny13** [comment]
Pushing this back into the backlog for now. This didnt turn out to be a straightforward implementation and its high priority. Some possible implementations after spiking:

- move the configs out of meltano.yml and into an env variable so we can make the date dynamic. This needs to be verified and bash command needs to be created to generate that dynamic date.
- Add custom Airflow dag that builds the tables view and exports it prior to running `meltano run` command

**Comment by pnadolny13** [comment]
Consider https://github.com/femueller/cloud-ip-ranges/issues/1 where https://www.microsoft.com/en-us/download/confirmation.aspx?id=56519 seems to pull the latest. From https://github.com/femueller/cloud-ip-ranges.

**Comment by pnadolny13** [comment]
This returns the correct date `$(date -d "@$(($(date -d "2023-02-27" +%s) + $(($(($(($(date +%s) - $(date -d "2023-02-27" +%s))) / 86400)) / 14)) * 1209600))" "+%Y%m%d")`
